### PR TITLE
Enabled server test to run when pushing to or making pull_request to master branch

### PR DESCRIPTION
**Pull request by devYaoYH**
- Added `firebase-tools` to server dev dependencies
- Run `yarn test` with `firebase emulators:exec "yarn test"` command

Known issue #100 lengthens server test by ~1min

### CODE REVIEW

File: server/tests/routes.test.js [hunk 1]
```diff
@@ -80,6 +81,11 @@ describe("API POST Data", () => {
     });
     expect(res.statusCode).toEqual(CONSTANTS.HTTP_SUCCESS);
     expect(res.text).toEqual("TEST_ORDER");
+    // Be nice and delete this order to avoid side effects
+    const addedOrders = await ordersCollection
```
Comment (BhavyaLight) [inline server/tests/routes.test.js:85]: why is this needed?

Comment (devYaoYH) [reply server/tests/routes.test.js:85]: Currently we are doing:
1) Clear DB in yarn pretest script
2) Populate DB in jest globalSetup script
3) Run tests

OK:
`DB.test.js` -> `routes.test.js`
Problem:
`routes.test.js` -> `DB.test.js` as we are adding an item to Orders Collection in one of our tests in `routes.test.js`

Solutions:
1) Avoid side effects by dropping this document after we assert it has been successfully added
2) Refactor code such that we clear DB and re-populate in jest `beforeAll` for each test suite

For now, I've taken the earlier approach

Comment (devYaoYH) [reply server/tests/routes.test.js:85]: actually, interesting point on test ordering, raised as issue in #105, the referenced approach seems feasible, might be a better way than the current fix

Comment (BhavyaLight) [reply server/tests/routes.test.js:85]: SGTM! 
2# seems like a better approach in long run! Perhaps just copy over the DB from DB.test.js into a new one in beforeAll and then delete it in AfterAll. The items stay neat then.
